projects/packages/forms/src/contact-form/class-contact-form-field.php

```diff
@@ -817,19 +817,75 @@ private function render_file_field( $id, $label, $class, $required, $required_fi
+				// translators: %s is the formatted maximum file size.
+				'fileTooLarge'       => sprintf( __( 'File is too large. Maximum allowed size is %s.', 'jetpack-forms' ), size_format( $max_file_size ) ),
+				'invalidType'        => __( 'This file type is not allowed', 'jetpack-forms' ),
+			),
+			'maxUploadSize'  => $max_file_size,
+			'endpoint' => rest_url( 'wpcom/v2/unauth-file-upload' ),
+			'wp_nonce'       => wp_create_nonce( 'wp_rest' ),
+			'jp_nonce'       => wp_create_nonce( 'jetpack_file_upload_jetpack-form' ),
+		);
 
-		// Localize the script only once.
-		$this->localize_file_field_script();
+		wp_interactivity_state( 'jpDropZone', $global_state );
 
-		return $field;
+		$context = array(
+			'isDropping' => false,
+			'files'      => array(),
+			'hasFiles'   => false,
+		);
+
+		$field  = $this->render_label( 'file', $id, $label, $required, $required_field_text );
+		ob_start();
+		?>
+<div
+	data-wp-interactive="jpDropZone"
+	<?php echo get_block_wrapper_attributes(); ?>
+	<?php echo wp_interactivity_data_wp_context( $context ); ?>
+	data-wp-on--dragover="actions.dragOver"
+	data-wp-on--dragleave="actions.dragLeave"
+	data-wp-on--mouseleave="actions.dragLeave"
+	data-wp-on--drop="actions.fileDropped"
+>
+	<div class="jetpack-form-file-field__dropzone" data-wp-class--is_dropping="context.isDropping"  >
+		<div class="jetpack-form-file-field__dropzone-inner" data-wp-on--click="actions.openFilePicker"></div>
+		<a href="#" class="wp-block-button__link wp-element-button"><?php esc_html_e( 'Select a file', 'jetpack-forms' ); ?></a>
+		<span class="jetpack-form-file-field__short"><?php esc_html_e( '....or drag and drop a file.', 'jetpack-forms' ); ?></span>
+		<input id="<?php echo esc_attr( $id ); ?>" type="file" class="jetpack-form-file-field" data-wp-on--change="actions.fileAdded" />
+	</div>
+	<div class="jetpack-form-file-field__preview-wrap" data-wp-class--is-active="context.hasFiles">
+		<template data-wp-each--file="context.files" data-wp-key="context.file.id">
+			<div class="jetpack-form-file-field__preview">
+				<div class="jetpack-form-file-field__progress" data-wp-bind--data-progress-id='context.file.id' data-wp-class--is-complete="context.file.hasToken" style="--progress: 3%;" role="progressbar" aria-valuenow="10" aria-valuemin="0" aria-valuemax="100"></div>
+				<input type="hidden" name="<?php echo esc_attr( $id ) ?>_token[]" data-wp-bind--value='context.file.token' value="">
+				<div class="jetpack-form-file-field__image" data-wp-style--background-image="context.file.url" ></div>
+				<div class="jetpack-form-file-field__file-wrap">
+					<span class="jetpack-form-file-field__file-name" data-wp-text="context.file.name"></span>
+					<span class="jetpack-form-file-field__file-size" data-wp-text="context.file.formattedSize"></span>
+				</div>
+				<a href="#" class="jetpack-form-file-field__remove" data-wp-bind--data-id='context.file.id' aria-label="Remove file" data-wp-on--click="actions.removeFile">Remove</a>
```

Comment:
Same here. 

It is also missing the `Remove` translation.

By the way, I see that you are serializing the translation of `Remove` to the frontend. If the translation is static in the DOM, it is not necessary to serialize it. That is, you can remove it from the `$i18n` array. Same for the other static translations.

Reply:
This was resolved by @enejb.